src/BrazeKit-dev.js

```diff
@@ -66,32 +66,92 @@ var constructor = function () {
         dob: 'setDateOfBirth',
     };
 
+    var bundleProductsWithEvent = false;
+
+    // A purchase event can either log a single event with all products
+    // or multiple purchase events (one per product)
     function logPurchaseEvent(event) {
         var reportEvent = false;
-        if (event.ProductAction.ProductList) {
-            event.ProductAction.ProductList.forEach(function (product) {
-                if (product.Attributes == null) {
-                    product.Attributes = {};
+
+        if (bundleProductsWithEvent) {
+            reportEvent = logSinglePurchaseEventWithProducts(event);
+        } else {
+            reportEvent = logPurchaseEventPerProduct(event);
+        }
+        return reportEvent === true;
+    }
+
+    function logSinglePurchaseEventWithProducts(event) {
+        var quantity = 1;
+        var eventAttributes = mergeObjects(event.EventAttributes, {
+            products: [],
+        });
+        if (event.ProductAction.TransactionId) {
+            eventAttributes['Transaction Id'] =
+                event.ProductAction.TransactionId;
+        }
+
+        if (event.ProductAction.ProductList.length) {
+            event.ProductAction.ProductList.forEach(function(_product) {
+                var product = getSanitizedCustomProperties(_product);
+
+                var productName;
+                if (forwarderSettings.forwardSkuAsProductName === 'True') {
+                    productName = _product.Sku;
+                } else {
+                    productName = _product.Name;
                 }
-                product.Attributes['Sku'] = product.Sku;
+                product.Name = getSanitizedValueForAppboy(productName);
+
+                eventAttributes.products.push(product);
+            });
+        }
+
+        kitLogger(
+            'appboy.logPurchase',
```

Comment:
Same here. We should make this a constant`

Reply:
Agree it should have been a constant but this is supposed to be temporary, and I consider it out of scope also.